src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -373,25 +374,20 @@ private async Task<int> Start()
 
             monitorTask.Start();
 
-            while(!shouldExit.WaitOne(250))
+            
+            while (!shouldExit.WaitOne(250) && !(!Console.IsInputRedirected && Console.KeyAvailable)) { }
```

Comment:
I'm not super familiar with this code, but doesn't this new code only handle one key press now? It used to have a loop within a loop, now it just loops once until a key is pressed and returns. Is that intended?

Reply:
Yup, looks like it lost the outer loop with this refactor.

This should do something like this:

```csharp
while (!shouldExit)
{
	if (!redirected && keyAvailable) { /* check 3 keys */ }
}

StopMonitor();
return ReturnCode.Ok;
```

Reply:
I added the outer loop.